src/service/video-manager-impl.js

```diff
@@ -132,9 +133,12 @@ class VideoEntry {
     /** @private {boolean} */
     this.userInteracted_ = false;
 
-    /** @const @private {!../service/vsync-impl.Vsync} */
+    /** @private @const {!../service/vsync-impl.Vsync} */
     this.vsync_ = vsyncFor(win);
 
+    /** @private @const {!../service/viewer-impl.Viewer} */
+    this.viewer_ = viewerFor(win);
```

Comment:
`viewer.isVisible()` can be called several times per video, might be worth caching a reference here.